core/src/test/java/org/elasticsearch/indices/recovery/IndexPrimaryRelocationIT.java

```diff
@@ -0,0 +1,234 @@
+ * under the License.
+ */
+
+package org.elasticsearch.indices.recovery;
+
+import org.elasticsearch.action.admin.indices.stats.IndicesStatsResponse;
+import org.elasticsearch.action.index.IndexRequestBuilder;
+import org.elasticsearch.cluster.routing.allocation.command.MoveAllocationCommand;
+import org.elasticsearch.common.lease.Releasable;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.index.IndexService;
+import org.elasticsearch.index.shard.IndexShard;
+import org.elasticsearch.index.shard.IndexShardState;
+import org.elasticsearch.index.shard.ShardId;
+import org.elasticsearch.index.store.Store;
+import org.elasticsearch.indices.IndicesService;
+import org.elasticsearch.test.ESIntegTestCase;
+import org.elasticsearch.test.ESIntegTestCase.ClusterScope;
+import org.elasticsearch.test.ESIntegTestCase.Scope;
+import org.elasticsearch.test.disruption.BlockClusterStateProcessing;
+import org.elasticsearch.test.disruption.MultiServiceDisruptionScheme;
+import org.elasticsearch.test.disruption.ServiceDisruptionScheme;
+import org.junit.Before;
+
+import java.util.concurrent.ExecutionException;
+
+import static org.elasticsearch.common.settings.Settings.settingsBuilder;
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertAcked;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.notNullValue;
+
+/**
+ * This class tests 3 scenarios for primary relocation (each a test method)
+ */
+@ClusterScope(scope = Scope.TEST, numDataNodes = 0)
+public class IndexPrimaryRelocationIT extends ESIntegTestCase {
+
+    private static final String INDEX_NAME = "test-idx-1";
+    private static final String INDEX_TYPE = "test-type-1";
+    private static final int MIN_DOC_COUNT = 500;
+    private static final int MAX_DOC_COUNT = 1000;
+    private static final int SHARD_COUNT = 1;
+    private static final int REPLICA_COUNT = 0;
+
+    private String nodeA = null;
+    private String nodeB = null;
+
+    private String nodeAId = null;
+    private String nodeBId = null;
+
+    private IndexShard nodeAShard = null;
+
+    private BlockClusterStateProcessing blockClusterStateProcessingOnNodeA;
+    private BlockClusterStateProcessing blockClusterStateProcessingOnNodeB;
+    private ServiceDisruptionScheme blockClusterStateProcessingOnBothNodes;
+
+    @Override
+    @Before
+    public void setUp() throws Exception {
+        super.setUp();
+
+        logger.info("--> start master node");
+        internalCluster().startMasterOnlyNode(Settings.EMPTY);
+        logger.info("--> start node A");
+        nodeA = internalCluster().startDataOnlyNode(Settings.EMPTY);
+
+        createAndPopulateIndex(INDEX_NAME, 1, SHARD_COUNT, REPLICA_COUNT);
+
+        logger.info("--> start node B");
+        nodeB = internalCluster().startDataOnlyNode(Settings.EMPTY);
+        ensureGreen();
+
+        client().admin().cluster().prepareState().get().getState().getNodes().forEach(node -> {
```

Comment:
you can do : `internalCluster().getInstance(ClusterService.class, nodeA).localNode().id();`